snap/info.go

```diff
@@ -799,6 +799,13 @@ func ReadInfo(name string, si *SideInfo) (*Info, error) {
 	}
 
 	st, err := os.Stat(MountFile(name, si.Revision))
+	if os.IsNotExist(err) {
+		// This can happen when "snap try" mode snap is moved around. The mount
+		// is still in place (it's a bind mount, it doesn't care about the
+		// source moving) but the symlink in /var/lib/snapd/snaps is now
+		// dangling.
+		return nil, &NotFoundError{Snap: name, Revision: si.Revision}
```

Comment:
this makes snapstate happy, otoh we lose information about exactly which bit of the snap is missing (before it was used only for missing snap.yaml)

Reply:
Correct. 

Do you think I should wrap the original error with more descriptive message in each of the cases in this function?

Reply:
I don't think we want the original message but maybe  something in parenthesis with the considered path:  ($path not found) or ($path does not exist)

Reply:
Shall we extend NotFoundError with a message that clarifies _what_ is not found precisely?

Reply:
as I said I would extend it I think to have an optional Path attribute, that registers the thing we could not find in the filesystem

Reply:
& construct the extra bits of the message with that

Reply:
I'll do just that, thank you!